### ISSUE TITLE
Medium not working thru revive

### ISSUE THREAD

**Issue by Dekor47** [LibreVR/Revive#922]
HI,

So currently im having a hard time running Occulus medium, every time at launch i get an instant "fatal error" MSG with a blank screen on my desktop. It looks to be just a snap shot of the main medium menu. While in VR I cant see the usual application loading screen but i can hear the loading music. also my controllers seem to phase thru selections, and become unresponsive. Usually pressing the home button on and off does the trick, but im not sure if these 2 issues are related in anyway but its weird since i wasent experiencing any of this yesterday and from what i can see nothing has updated except steam vr. 

Just before making this post i had updated revive and got a new msg saying "Unable to initialize the Oculus Rift"

i have seen similar issues before but since steamvr just updated im wondering if this could be the issues.

ive reinstalled revive twice. once to a previous version and 2nd time i just reinstalled the newest version.
ive reinstalled medium
ive disabled steamvr beta
ive updated my drivers to 390.77

im on a Nvidia geforce GTX 970

**Comment by Armada651** [comment]
Can you try switching to the openvr-inputemulator-temporary branch of SteamVR?

**Comment by Armada651** [comment]
That branch uses an old build of SteamVR, the issue you're describing was in the beta and I hope they didn't just push it to the stable.

**Comment by 3dbumper** [comment]
> Can you try switching to the openvr-inputemulator-temporary branch of SteamVR?

I am having the same issue. Can you expand upon your work-around? I'm not familiar with this procedure.

**Comment by cartoonmonkey** [comment]
Windows Mixed Reality (Odyssey) user here. The openvr-inputemulator-temporary branch of SteamVR fails to even recognize my headset, so that's no solution for me.

**Comment by Armada651** [comment]
@cartoonmonkey You'll need to wait until Valve fixes the issue or until I complete native WMR support in Revive.

**Comment by mm373** [comment]
Are you saying we need to install openvr inputemulator? I tried that and am also trying to start in non beta as well as fresh install of steam and revive with no luck.

**Comment by Armada651** [comment]
No.... I said you needed to switch to a different branch, it's right there in the list next to the beta.

**Comment by mm373** [comment]
Is that for the beta that ended jan 22? I don't see that option in steamvr


Sent via the Samsung Galaxy S7, an AT&T 4G LTE smartphone
-------- Original message --------From: Jules Blok <notifications@github.com> Date: 2/19/18  6:30 AM  (GMT-06:00) To: LibreVR/Revive <Revive@noreply.github.com> Cc: mm373 <mbroden@gmail.com>, Comment <comment@noreply.github.com> Subject: Re: [LibreVR/Revive] Medium "Fatal error" On start up (#922) 
No.... I said you needed to switch to a different branch, it's right there in the list next to the beta.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/LibreVR/Revive","title":"LibreVR/Revive","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/LibreVR/Revive"}},"updates":{"snippets":[{"icon":"PERSON","message":"@Armada651 in #922: No.... I said you needed to switch to a different branch, it's right there in the list next to the beta."}],"action":{"name":"View Issue","url":"https://github.com/LibreVR/Revive/issues/922#issuecomment-366679859"}}}

**Comment by Armada651** [comment]
![medium](https://user-images.githubusercontent.com/2028870/36393616-9933ab96-15b0-11e8-9ac2-22e830fefb33.png)

**Comment by Armada651** [comment]
Can someone test the beta branch of SteamVR? It should be fixed there also.

**Comment by AzraelXS** [comment]
Wish I'd seen this earlier - Windows is downloading a new insider build now (after several hours of troubleshooting why I'm 3 builds behind), and it's late, so I probably won't be able to test it until I get off work tomorrow, but I'm all about helping test it for you. Might be able to check it before the PC reboots to finish the install.

And I'd like to re-iterate the sentiment of Dekor47 - Thank you very much for all you do.

**Comment by AzraelXS** [comment]
Unfortunately, the SteamVR Beta branch is a no-go. The openvr-inputemulator temporary branch works though, at least... temporarily, haha.

**Comment by Expicot** [comment]
Steam broke it again :(.

No version is working (beta, imputemulator, GDC2018, standard)... 
Arggh, that complicated world where things were working a day and no more the next day... and hopefully again the next next day but so far no Medium for me today.

**Comment by Expicot** [comment]
Somehow related : does someone knows how to 'backup' a given (working) SteamVR version ? That could be a 'lifeguard' when you have to rely on a software.

**Comment by Armada651** [comment]
@Expicot You need to disable the Steam Overlay: https://www.patreon.com/posts/how-to-fix-part-17220186

**Comment by Expicot** [comment]
Thanks for the hint but it was disabled. Previously it was working only on SteamVR beta, but since the latest update (version with the automatic SS setting)... it just crash at start.

**Comment by Armada651** [comment]
Next version of Revive will disable the code in Oculus Medium that makes it so sensitive to OpenGL errors. Should be out today or tomorrow.

**Comment by Expicot** [comment]
Great ! Thanks a lot.

**Comment by 3dbumper** [comment]
Thank you so much. I am demoing VR sculpting, using Medium, on the 28th and
was seriously sweating it.

**Comment by redcovert** [comment]
Just to confirm I tried the openvr-inputemulator-temporary branch and making sure I didn't have steam overlay on but it is not working for me either.  I will check back in a couple days and see if there have been any updates.  Windows 10/32 Gb/Nvidia 1080. Thanks.

**Comment by redcovert** [comment]
I see there was a "commit" to fix the issue for medium (hooray!!) but I am a github noob and have no clue how to get that installed if it is not included in the existing installer. What is the trick to get that commit installed?

**Comment by Armada651** [comment]
@redcovert I don't have a nightly build system so you'll have to wait until it is rolled out in the next release.

**Comment by 3dbumper** [comment]
@Armada651 - Do you have an idea as to when the next release will come?

**Comment by Armada651** [comment]
@3dbumper Right now: https://github.com/LibreVR/Revive/releases/tag/1.2.2

**Comment by 3dbumper** [comment]
@Armada651 Thank you very much. I really appreciate it.

**Comment by 3dbumper** [comment]
I apologize if this is the wrong place to ask this, but are there any tricks to getting the most performance out of Medium?  My GPU (1080) never got over 58% usage during  my last session, and I'm getting some frame rate issues/stutter.

**Comment by marcoleone** [comment]
@Armada651 Hi, I just updated to the latest release form your link and Medium is no working at all along with Quill that was working fine before the update.
Any idea on what could be wrong? Thank you!

**Comment by marcoleone** [comment]
Sorry, just realize I had to update the oculus app as well.. All working fine now.
Thank you so much for you work!

**Comment by redcovert** [comment]
Github Noon here - so just to confirm with the 1.2.2 zip you link to above... do you just unzip that into the existing revive folder?

**Comment by Expicot** [comment]
Latest official release is 1.2.1 right ?
Is it possible that performance is lower with Medium ? I have a lot of reprojection that I don't remember having saw in previous builds (GTX 1080Ti).

**Comment by kool601** [comment]
@redcovert You're downloading the source code, you need to download the executable file called "ReviveInstaller.exe".

**Comment by MatiasBrunacci** [comment]
It's happening again with Oculus Medium version 2.1
I had the same problem before with Medium version 2.0 and I solved it installing SteamVR Beta. But now it's not working again, even with SteamVR Beta installed.
